multimap/src/test/java/org/infinispan/multimap/impl/EmbeddedMultimapCacheWithDuplicatesTest.java

```diff
@@ -0,0 +1,91 @@
+package org.infinispan.multimap.impl;
+
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.multimap.api.embedded.EmbeddedMultimapCacheManagerFactory;
+import org.infinispan.multimap.api.embedded.MultimapCache;
+import org.infinispan.multimap.api.embedded.MultimapCacheManager;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.data.Person;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Test;
+
+import static org.infinispan.functional.FunctionalTestUtils.await;
+import static org.infinispan.multimap.impl.MultimapTestUtils.NAMES_KEY;
+import static org.infinispan.multimap.impl.MultimapTestUtils.OIHANA;
+import static org.infinispan.multimap.impl.MultimapTestUtils.RAMON;
+import static org.infinispan.multimap.impl.MultimapTestUtils.JULIEN;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@Test(groups = "functional", testName = "multimap.EmbeddedMultimapCacheWithDuplicatesTest")
+public class EmbeddedMultimapCacheWithDuplicatesTest extends SingleCacheManagerTest {
+    protected MultimapCache<String, Person> multimapCache;
+
+    @Override
+    protected EmbeddedCacheManager createCacheManager() throws Exception {
+        // start a single cache instance
+        EmbeddedCacheManager cm = TestCacheManagerFactory.createCacheManager(MultimapSCI.INSTANCE);
+        cm.defineConfiguration("test", new ConfigurationBuilder().build());
+        MultimapCacheManager multimapCacheManager = EmbeddedMultimapCacheManagerFactory.from(cm);
+        multimapCache = multimapCacheManager.get("test", true);
+        cm.getClassAllowList().addClasses(SuperPerson.class);
+        return cm;
+    }
+
+    public void testSupportsDuplicates() {
+        assertTrue(multimapCache.supportsDuplicates());
+    }
+
+    public void testPutDuplicates() {
+
+        await(multimapCache.put(NAMES_KEY, JULIEN)
```

Comment:
I am personally not a fan of the multiple chained stages for these tests, as I think it is a lot easier to understand with individual awaits. The individual awaits also make the stack trace a lot easier to parse when a failure happens. But I would say it is okay to integrate as it.